src/ui/popup.js

```diff
@@ -600,10 +619,9 @@ export default class Popup extends Evented {
 }
 
 function normalizeOffset(offset: ?Offset) {
-    if (!offset) {
-        return normalizeOffset(new Point(0, 0));
```

Comment:
Seemed like a strange case for recursion here

Reply:
I believe I've seen this style, especially in the symbol code, where it runs modified values through a different branch of the same function, but TBH I find it slightly hard to follow, so that if otherwise equivalent, I prefer your change 👍